janus_server/src/binary_utils.rs

```diff
@@ -223,7 +238,11 @@ where
         &common_options.database_password,
         &common_options.datastore_keys,
     )
-    .context("couldn't connect to database")?;
+    .context("couldn't create database connection pool")?;
+    datastore
+        .check_connection_pool()
+        .await
+        .context("couldn't connect to database")?;
```

Comment:
Now, all binaries will try to connect to the database at startup. This arguably makes the call to `Datastore::check_connection_pool` in the aggregator's `healthz/startup` endpoint redundant (it couldn't get so far as to serve that request if this call to `Datastore::check_connection_pool` hadn't already succeeded), so perhaps we can delete the `startup` endpoint.

Reply:
`janus_cli` doesn't want this since not all commands want a datastore connection at all, but we already plan to change `janus_cli` to not use the "standard server binary startup" codepath in #315, so I think it's OK to add this check for now.

I do think we should expose health/startup endpoints from all server components. Maybe these "z-page" (internal status) pages could be factored out to a common module, and maybe run on a different port (to ensure that they are never accessible from outside the cluster)? I think with the way that `/healthz` is currently implemented on the aggregator, once we're launched, anyone on the Internet could hit `foo.api.divviup.org/healthz` and see our internal healthz response. While that may not indicate much of a practical problem, I suspect we're going to add more status pages as we go, and some of those might contain information we would prefer not to be public.

Reply:
(all of the above, of course, doesn't require a change in this PR)

Reply:
I agree with both your points and I made a note in #36 about it.